File: src/main/java/com/agentstore/common/exception/client/ClientException.java
package com.agentstore.common.exception.client;

import com.agentstore.common.exception.BusinessException;
import com.agentstore.common.exception.constants.ErrorCode;

public abstract class ClientException extends BusinessException {
    protected ClientException(ErrorCode errorCode, Object... messageArguments) {
        super(errorCode, messageArguments);
    }

    protected ClientException(ErrorCode errorCode, Throwable cause, Object... messageArguments) {
        super(errorCode, cause, messageArguments);
    }
}

File: src/main/kotlin/com/agentstore/common/exception/client/ClientException.kt
package com.agentstore.common.exception.client

import com.agentstore.common.exception.BusinessException
import com.agentstore.common.exception.constants.ErrorCode

abstract class ClientException : BusinessException {
    protected constructor(errorCode: ErrorCode, vararg messageArguments: Any?) : super(
        errorCode,
        *messageArguments,
    )

    protected constructor(
        errorCode: ErrorCode,
        cause: Throwable,
        vararg messageArguments: Any?,
    ) : super(errorCode, cause, *messageArguments)
}
